handler/src/main/java/io/netty/handler/ssl/util/FingerprintTrustManagerFactory.java

```diff
@@ -75,20 +83,8 @@
 
     private static final Pattern FINGERPRINT_PATTERN = Pattern.compile("^[0-9a-fA-F:]+$");
     private static final Pattern FINGERPRINT_STRIP_PATTERN = Pattern.compile(":");
-    private static final int SHA1_BYTE_LEN = 20;
-    private static final int SHA1_HEX_LEN = SHA1_BYTE_LEN * 2;
 
-    private static final FastThreadLocal<MessageDigest> tlmd = new FastThreadLocal<MessageDigest>() {
-        @Override
-        protected MessageDigest initialValue() {
-            try {
-                return MessageDigest.getInstance("SHA1");
-            } catch (NoSuchAlgorithmException e) {
-                // All Java implementation must have SHA1 digest algorithm.
-                throw new Error(e);
-            }
-        }
-    };
+    private final FastThreadLocal<MessageDigest> tlmd = new FastThreadLocal<MessageDigest>();
```

Comment:
@chrisvest was right here... you need to override `initialValue()` here

Reply:
Since `FingerprintTrustManagerFactory` is now configured with a hash algorithm, `tlmd` needs to be initialized in the constructor. It used to be before. Then, `initialValue()` was removed because we wanted to create an instance of `MessageDigest` earlier in the constructor to check if the algorithm exists. If I understand correctly, we just want to revert the logic to the previous version. I am not sure if we need to create another `MessageDigest` in the constructor. Please let me know if so.

Reply:
> I am not sure if we need to create another MessageDigest in the constructor. Please let me know if so.

Well, looks like there is no harm if we do that. So, I'll add such a check if no objections.

Reply:
Since `tlmd` is an instance field, you can create the `FastThreadLocal` with an `initialValue` method that has all the logic necessary to create `MessageDigest` instances. I'd still check that the algorithm exists before creating the FTL, though, because each FTL instance will claim some static resources that we'd prefer not to leak.